.github/workflows/run-unit-tests.yml

```diff
@@ -35,13 +43,21 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
+
+      - name: Install poetry via pipx
+        run: pipx install poetry
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
-      - name: Install Poetry
-        run: curl -sSL https://install.python-poetry.org | python3 -
+          cache: 'poetry'
+
+      - name: Install Python dependencies using Poetry
+        run: poetry install --without evaluation
```

Comment:
this is part of `make build`

Reply:
I know which have been removed in previous commit. I think add `poetry install` here can speed up `actions` and looks like no other side effects so i added it here.

Reply:
Logically it wouldn't make sense to speed up... but it wouldn't harm either. I feel like we could remove `make build` completely from run-integration-tests.yml though.